src/services/services.ts

```diff
@@ -5762,6 +5773,62 @@ namespace ts {
             return forEach(diagnostics, diagnostic => diagnostic.category === DiagnosticCategory.Error);
         }
 
+        function getDependencies(fileName: string): DependencyInfo {
+            synchronizeHostData();
+            
+            let sourceFile = getValidSourceFile(fileName);
+            if (!sourceFile) {
+                return undefined;
+            }
+            
+            let resolver = program.getDiagnosticsProducingTypeChecker().getEmitResolver(sourceFile);
+            
+            let runtime: string[] = [];  
+            let compileTime: string[] = [];  
+            for (let node of sourceFile.amdDependencies) {  
+                runtime.push(node.path);  
+            }  
+            
+            function getModuleNameText(node: ImportDeclaration | ImportEqualsDeclaration |  ExportDeclaration): string {
+                let moduleName = getExternalModuleName(node);
+                return (<LiteralExpression>moduleName).text;                
+            }
+            
+            for (let node of sourceFile.statements) {
+                switch (node.kind) {  
+                    case SyntaxKind.ImportDeclaration:
+                        let importDeclaration = <ImportDeclaration>node;
+                        if (!importDeclaration.importClause || resolver.isReferencedAliasDeclaration(importDeclaration.importClause, /*checkChildren*/ true)) {

```

Comment:
you need to force a typecheck on the file to make sure this is set correctly. so you need to query for all semantic diagnostics.